src/transformers/models/phi/modeling_phi.py

```diff
@@ -0,0 +1,1072 @@
+    def __init__(self, config):
+        super().__init__(config)
+        self.model = PhiModel(config)
+        self.vocab_size = config.vocab_size
+        self.lm_head = nn.Linear(config.hidden_size, config.vocab_size, bias=True)
+
+        # Initialize weights and apply final processing
+        self.post_init()
+
+    # Copied from transformers.models.llama.modeling_llama.LlamaForCausalLM.get_input_embeddings
+    def get_input_embeddings(self):
+        return self.model.embed_tokens
+
+    # Copied from transformers.models.llama.modeling_llama.LlamaForCausalLM.set_input_embeddings
+    def set_input_embeddings(self, value):
+        self.model.embed_tokens = value
+
+    # Copied from transformers.models.llama.modeling_llama.LlamaForCausalLM.get_output_embeddings
+    def get_output_embeddings(self):
+        return self.lm_head
+
+    # Copied from transformers.models.llama.modeling_llama.LlamaForCausalLM.set_output_embeddings
+    def set_output_embeddings(self, new_embeddings):
+        self.lm_head = new_embeddings
+
+    # Copied from transformers.models.llama.modeling_llama.LlamaForCausalLM.set_decoder
+    def set_decoder(self, decoder):
+        self.model = decoder
+
+    # Copied from transformers.models.llama.modeling_llama.LlamaForCausalLM.get_decoder
+    def get_decoder(self):
+        return self.model
+
+    @add_start_docstrings_to_model_forward(PHI_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=CausalLMOutputWithPast, config_class=_CONFIG_FOR_DOC)
+    def forward(
+        self,
+        input_ids: torch.LongTensor = None,
+        attention_mask: Optional[torch.Tensor] = None,
+        position_ids: Optional[torch.LongTensor] = None,
+        past_key_values: Optional[List[torch.FloatTensor]] = None,
+        inputs_embeds: Optional[torch.FloatTensor] = None,
+        labels: Optional[torch.LongTensor] = None,
+        use_cache: Optional[bool] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+    ) -> Union[Tuple, CausalLMOutputWithPast]:
+        r"""
+        Args:
+            labels (`torch.LongTensor` of shape `(batch_size, sequence_length)`, *optional*):
+                Labels for computing the masked language modeling loss. Indices should either be in `[0, ...,
+                config.vocab_size]` or -100 (see `input_ids` docstring). Tokens with indices set to `-100` are ignored
+                (masked), the loss is only computed for the tokens with labels in `[0, ..., config.vocab_size]`.
+
+        Returns:
+
+        Example:
+
+        ```python
+        >>> from transformers import AutoTokenizer, PhiForCausalLM
+
+        >>> model = PhiForCausalLM.from_pretrained("susnato/phi-1_dev")
+        >>> tokenizer = AutoTokenizer.from_pretrained("susnato/phi-1_dev")
+
+        >>> prompt = "Hey, are you conscious? Can you talk to me?"
```

Comment:
Let's avoid example where we talk about consciousness for models

Reply:
In the above example we use 
`model = PhiForCausalLM.from_pretrained("susnato/phi-1_dev")`, where PhiForCausalLM gives an import error.

In the official docs we're using 
`model = AutoModelForCausalLM.from_pretrained("microsoft/phi-1_5", trust_remote_code=True)`.

Are both of these functions of loading the model the same? I only ask because I know with previous versions, you could not fine tune the model, so I wanna make sure fine tuning is now corrected using both function calls.

Reply:
Hi @bnicholl, if you want to use `PhiForCausalLM` right now you need to checkout my branch (`susnato:phi`) and then install from there, otherwise wait for this PR to get merged. :)

> Are both of these functions of loading the model the same? I only ask because I know with previous versions, you could not fine tune the model, so I wanna make sure fine tuning is now corrected using both function calls.

Once this PR is merged it will load the same model and yes you can fine-tune like any other models from this library!